adb_graphics/datahandler/grib.py

```diff
@@ -451,6 +451,36 @@ def grid_info(self):
 
         return grid_info
 
+    def supercooled_liquid_water(self, values, **kwargs) -> np.ndarray:
+
+        # pylint: disable=unused-argument
+
+        '''
+        Generates a field of Supercooled Liquid Water
+        '''
+
+        ps = self.values(name='pres', level='sfc') * 100. # convert back to Pa
+        p = self.values(name='pres', level='ua', one_lev=False)
+        t = self.values(name='temp', level='ua', one_lev=False)
+        cmr = self.values(name='clwmr', level='ua', one_lev=False)
+        rmr = self.values(name='rwmr', level='ua', one_lev=False)
+
+        g = 9.81 # gravity
+        slw = ps * 0.
+
+        nlevs = np.shape(p)[0]
+        for n in range(nlevs):
+            if n == 0:
+                dp = 2 * (ps[:, :] - p[n, :, :])
+                Pm = ps - dp
+            else:
+                dp = 2 * (Pm[:, :] - p[n, :, :])
+                Pm = Pm - dp
```

Comment:
A comment on what the next two lines are dong would be helpful.

I'm guessing "Set upper boundary of slw to 0 when temperature is above freezing."

I'm not even sure how to guess on what the next line is doing. Scaling it based on the pressure change?

Reply:
added comments